web/source/keymanweb.js

```diff
@@ -3089,39 +3087,36 @@
     var kbdLang = kbdStub['KL'];
     var kbdName = kbdStub['KN'];
 
-    if(typeof(kbdStub.asyncLoader.callback) == "function") {
-      // Add a handler for cases where the new <script> block fails to load.
-      Lscript.addEventListener('error', function() {
+
+    // Add a handler for cases where the new <script> block fails to load.
+    Lscript.addEventListener('error', function() {
+      if(kbdStub.asyncLoader.timer !== null) {
         // Clear the timeout timer.
         window.clearTimeout(kbdStub.asyncLoader.timer);
         kbdStub.asyncLoader.timer = null;
+      }
 
+      // Do not output error messages when embedded.  (KMEA/KMEI)
+      if(!keymanweb.embedded) {
         // We already know the load has failed... why wait?
-        kbdStub.asyncLoader.callback('Cannot find the ' + kbdName + ' keyboard for ' + kbdLang + '!');
-        console.log('Error:  cannot find the', kbdName, 'keyboard for', kbdLang, 'at', kbdFile + '!');
+        kbdStub.asyncLoader.callback('Cannot find the ' + kbdName + ' keyboard for ' + kbdLang + '.');
+        console.log('Error:  cannot find the', kbdName, 'keyboard for', kbdLang, 'at', kbdFile + '.');
+      }
+      kbdStub.asyncLoader = null;
+    }, true);
 
-        kbdStub.asyncLoader = null;
-      }, true);
-    }
     
     // The load event will activate a newly-loaded keyboard if successful and report an error if it is not.
     Lscript.addEventListener('load', function() {
-      if(typeof(kbdStub.asyncLoader.callback) == "function") {
+      if(kbdStub.asyncLoader.timer !== null) {
         // Clear the timeout timer.
         window.clearTimeout(kbdStub.asyncLoader.timer);
         kbdStub.asyncLoader.timer = null;
       }
 
       // To determine if the load was successful, we'll need to check the keyboard array for our desired keyboard.
-      //FIXME:
       // Test if keyboard already loaded
-      var kbd, Li;
-      for(Li=0; Li<keymanweb._Keyboards.length; Li++) {
-        if(kbdStub['KI'] == keymanweb._Keyboards[Li]['KI']) {
-          kbd = keymanweb._Keyboards[Li];
-        } 
-      }
-
+      var kbd = keymanweb._getKeyboardInternal(kbdStub['KI']), Li;
       if(kbd != undefined) {  // Is cleared upon a successful load.
```

Comment:
`null` (or `if(kbd)`) rather than `undefined`? `_getKeyboardInternal` returns `null` on not found.